collector/datacollector/datacollector.go

```diff
@@ -58,13 +67,53 @@ func main() {
 		glog.Exitf("Unable to obtain Log metadata: %s", err)
 	}
 
+	var ca *certgen.CA
+	if *addChainPeriod > 0 {
+		// TODO(katjoyce): Add support for other key encodings and
+		// generally improve key management here.
+		signingCertPEM, err := ioutil.ReadFile(*signingCertFile)
+		if err != nil {
+			glog.Exitf("Error reading signing cert from %s: %s", *signingCertFile, err)
+		}
+		signingCert, err := x509util.CertificateFromPEM(signingCertPEM)
+		if err != nil {
+			glog.Exitf("Error parsing signing cert: %s", err)
+		}
+
+		signingKeyPEM, err := ioutil.ReadFile(*signingKeyFile)
+		if err != nil {
+			glog.Exitf("Error reading signing key from %s: %s", *signingKeyFile, err)
+		}
+		signingKey, err := pem.UnmarshalPrivateKey(string(signingKeyPEM), "")
+		if err != nil {
+			glog.Exitf("Error parsing signing key: %s", err)
+		}
+
+		certConfig := certgen.CertificateConfig{
+			SubjectCommonName:         "flowers-to-the-world.com",
+			SubjectOrganization:       "Google",
+			SubjectOrganizationalUnit: "Certificate Transparency",
+			SubjectLocality:           "London",
+			SubjectCountry:            "GB",
+			SignatureAlgorithm:        x509.SHA256WithRSA,
+			DNSPrefix:                 l.Name,
+		}
+
+		ca = &certgen.CA{
+			SigningCert: signingCert,
+			SigningKey:  signingKey,
+			CertConfig:  certConfig,
+		}
```

Comment:
I would add the `certgen.CA` to the `Config` struct, it looks a bit weird to pass it as a separate thing to the collector? In particular, because it's allowed to be `nil`, as opposed to the HTTP client or the storage?

Reply:
Sure, done.